provider/ec2/environ.go

```diff
@@ -2590,7 +2654,14 @@ func (e *environ) SetCloudSpec(ctx stdcontext.Context, spec environscloudspec.Cl
 
 	var err error
 	e.ec2Client, err = e.ec2ClientFunc(ctx, spec, WithHTTPClient(httpClient.Client()))
-	return err
+	if err != nil {
+		return errors.Annotate(err, "creating aws ec2 client")
+	}
+	e.iamClient, err = e.iamClientFunc(ctx, spec, WithHTTPClient(httpClient.Client()))
+	if err != nil {
+		return errors.Annotate(err, "creating aws iam client")
```

Comment:
It would be better to rework the ec2ClientFUnc and iamClientFunc to take cfg as an arg, not cloud spec. 
Otherwise we are creating cfg from cloudspec twice. We should just construct cfg here once and pass that in to both funcs.